src/Stratis.Bitcoin.Features.Consensus/CoinViews/CachedCoinView.cs

```diff
@@ -170,7 +171,7 @@ private CachedCoinView(IDateTimeProvider dateTimeProvider, ILoggerFactory logger
             var outputs = new UnspentOutputs[txIds.Length];
             var miss = new List<int>();
             var missedTxIds = new List<uint256>();
-            using (await this.lockobj.LockAsync().ConfigureAwait(false))
+            using (await this.lockobj.LockAsync(ct).ConfigureAwait(false))
```

Comment:
Why did you do that? Is there a deeper thought behind that or is it just random improvement? You do realise that you introduced exception to the method that is not handled anywhere?